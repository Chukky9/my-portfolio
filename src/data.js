import javascript from './images/icons/javascript.png';
import laravel from './images/icons/laravel-48.png';
import nodeJs from './images/icons/node-js.png';
import react from './images/icons/react.png';
import vueJs from './images/icons/vue-js.png';

import calculator from './images/calculator.png'
import bookLibrary from './images/book-library.png';
import ticTacToe from './images/tic-tac-toe.png';

export const logos = [
    { name: 'javascript', logo: javascript },
    { name: 'react', logo: react },
    { name: 'vueJs', logo: vueJs },
    { name: 'nodeJs', logo: nodeJs },
    { name: 'laravel', logo: laravel },
]

export const projects = [
    { name: 'Calculator App', link: 'https://chuks-calculator.herokuapp.com/', image: calculator },
    { name: 'Book Library App', link: 'ttps://chuks-book-library.herokuapp.com/', image: bookLibrary },
    { name: 'Tic-Tac-Toe App', link: 'https://ttt-9.herokuapp.com/', image: ticTacToe },
]

export const contactLinks = [
    { name: 'Send a mail', link: 'mailto:chukkyalozie@gmail.com', icon: 'fa fa-at', target: null },
    { name: 'Call me', link: 'tel:+2348067877325', icon: 'fa fa-mobile', target: null },
    { name: 'Linkedin', link: 'https://linkedin.com/in/chukwuemekaalozie', icon: 'fa fa-linkedin-square', target: '_blank' },
    { name: 'GitHub', link: 'https://github.com/Chukky9', icon: 'fa fa-github', target: '_blank' },
]