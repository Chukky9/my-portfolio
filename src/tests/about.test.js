import { screen, render } from '@testing-library/react';
import About from '../components/about.jsx';
import { logos, contactLinks } from '../data';

test('renders about page with images and links successfully', () => {
    const { container } = render(<About/>)
    const logo = screen.getAllByRole('img')
    const links = screen.getAllByRole('link')

    expect(logo).toHaveLength(logos.length)
    expect(links).toHaveLength(contactLinks.length)
    links.forEach(link => {
        expect(link).toBeEnabled()
    })
    expect(container).toMatchSnapshot()
})