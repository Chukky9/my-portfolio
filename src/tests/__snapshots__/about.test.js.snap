// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders about page with images and links successfully 1`] = `
<div>
  <section
    class="contact-section"
  >
    <div
      class="contact-section-header"
    >
      <div
        class="tech-stack"
      >
        <h3>
          Tech Stack
        </h3>
        <div
          class="tech-logos"
        >
          <div
            class="logo-div"
          >
            <img
              alt="javascript"
              class="img-responsive logo-images"
              src=""
            />
          </div>
          <div
            class="logo-div"
          >
            <img
              alt="react"
              class="img-responsive logo-images"
              src=""
            />
          </div>
          <div
            class="logo-div"
          >
            <img
              alt="vueJs"
              class="img-responsive logo-images"
              src=""
            />
          </div>
          <div
            class="logo-div"
          >
            <img
              alt="nodeJs"
              class="img-responsive logo-images"
              src=""
            />
          </div>
          <div
            class="logo-div"
          >
            <img
              alt="laravel"
              class="img-responsive logo-images"
              src=""
            />
          </div>
        </div>
      </div>
      <p>
        How do you like your websites?
      </p>
    </div>
    <div
      class="contact-links"
    >
      <a
        class="contact-details"
        href="mailto:chukkyalozie@gmail.com"
      >
        <i
          class="fa fa-at"
        />
         
        Send a mail
      </a>
      <a
        class="contact-details"
        href="tel:+2348067877325"
      >
        <i
          class="fa fa-mobile"
        />
         
        Call me
      </a>
      <a
        class="contact-details"
        href="https://linkedin.com/in/chukwuemekaalozie"
        target="_blank"
      >
        <i
          class="fa fa-linkedin-square"
        />
         
        Linkedin
      </a>
      <a
        class="contact-details"
        href="https://github.com/Chukky9"
        target="_blank"
      >
        <i
          class="fa fa-github"
        />
         
        GitHub
      </a>
    </div>
  </section>
</div>
`;
